.github/workflows/ci.yml

```diff
@@ -0,0 +1,28 @@
+name: CI
+on:
+  pull_request:
+permissions:
+  contents: read
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - name: ci/Checkout code
+      uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+    - uses: actions/setup-python@db9987b4c1f10f0404fa60ee629f675fafbd6763 # v4.6.0
+      with:
+        python-version: '3.9'
+    - name: ci/Build
+      run: |
+        make python-deps
+        make html
+    - name: ci/Persist docs artifacts
+      uses: actions/upload-artifact@65d862660abb392b8c4a3d1195a2108db131dd05 # v3.1.2
+      with:
+        name: docs
+        path: build/html
+    - name: ci/Persist doc logs artifacts
+      uses: actions/upload-artifact@65d862660abb392b8c4a3d1195a2108db131dd05 # v3.1.2
+      with:
```

Comment:
Add retention here for the CI builds . Do we actually need those uploaded ? Aren't they visible on the console ?

Reply:
There is retention. It is the default(90d), and this was the way that this pipeline used to run from ever. 
Sure we can change this. Kindly define a proposed retention . 


-- 
The way `workflow_run:` works, is that we completely seperate the build process and the deploy process, to be able to safely run with inheriting secrets to the CD pipeline. 

In order to do that, we need a way to connect the calling pipeline (CI_MASTER) with the actual pipeline that deploys the build to Amazon S3. This is performed via upload-artifacts.  This is how we safely share data between these pipelines. 

The privileged pipeline containing AWS Secrets, does not build anything. for security reasons.

Reply:
@toninis  Any objections merging this as is?